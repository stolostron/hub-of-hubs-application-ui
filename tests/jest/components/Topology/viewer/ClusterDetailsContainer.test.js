// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
"use strict";

import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import ClusterDetailsContainer from "../../../../../src-web/components/Topology/viewer/ClusterDetailsContainer";

const locale = "US-en";
window.open = () => {}; // provide an empty implementation for window.open

describe("ClusterDetailsContainer with no clusters", () => {
  const mockData = {
    clusters: [],
    clusterDetailsContainerControl: {
      clusterDetailsContainerData: {
        page: 1,
        startIdx: 0,
        clusterSearchToggle: false,
        isSelectOpen: false,
        expandSectionToggleMap: new Set()
      },
      handleClusterDetailsContainerUpdate: jest.fn()
    }
  };
  it("renders as expected", () => {
    const component = renderer.create(
      <ClusterDetailsContainer
        clusterList={mockData.clusters}
        locale={locale}
        clusterDetailsContainerControl={mockData.clusterDetailsContainerControl}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe("ClusterDetailsContainer with some clusters", () => {
  const mockData = {
    clusters: [
      {
        cpu: "12",
        memory: "23308Mi",
        name: "argo-fxiang-eks",
        namespace: "argo-fxiang-eks",
        status: "ok"
      },
      {
        allocatable: { cpu: "11580m", memory: "20056Mi" },
        capacity: { cpu: "12", memory: "23308Mi" },
        metadata: {
          name: "fxiang-eks",
          namespace: "fxiang-eks"
        },
        status: "ok"
      },
      {
        allocatable: { cpu: "33", memory: "137847Mi" },
        capacity: { cpu: "36", memory: "144591Mi" },
        consoleURL:
          "https://console-openshift-console.apps.vbirsan1-remote.dev06.red-chesterfield.com",
        metadata: {
          name: "vbirsan1-remote",
          namespace: "vbirsan1-remote"
        },
        status: "ok"
      }
    ],
    clusterDetailsContainerControl: {
      clusterDetailsContainerData: {
        page: 1,
        startIdx: 0,
        clusterSearchToggle: false,
        isSelectOpen: false,
        expandSectionToggleMap: new Set()
      },
      handleClusterDetailsContainerUpdate: jest.fn()
    }
  };
  it("renders as expected", () => {
    const component = renderer.create(
      <ClusterDetailsContainer
        clusterList={mockData.clusters}
        clusterID="TestClusterID"
        locale={locale}
        clusterDetailsContainerControl={mockData.clusterDetailsContainerControl}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe("ClusterDetailsContainer test functions", () => {
  const mockData = {
    clusters: [
      {
        allocatable: { cpu: "11580m", memory: "20056Mi" },
        capacity: { cpu: "12", memory: "23308Mi" },
        metadata: {
          name: "fxiang-eks",
          namespace: "fxiang-eks"
        },
        status: "ok"
      },
      {
        allocatable: { cpu: "33", memory: "137847Mi" },
        capacity: { cpu: "36", memory: "144591Mi" },
        consoleURL:
          "https://console-openshift-console.apps.vbirsan1-remote.dev06.red-chesterfield.com",
        metadata: {
          name: "vbirsan1-remote",
          namespace: "vbirsan1-remote"
        },
        status: "ok"
      }
    ],
    clusterDetailsContainerControl: {
      clusterDetailsContainerData: {
        page: 1,
        startIdx: 0,
        clusterSearchToggle: false,
        isSelectOpen: false,
        expandSectionToggleMap: new Set()
      },
      handleClusterDetailsContainerUpdate: jest.fn()
    }
  };

  let wrapper;
  beforeEach(
    () =>
      (wrapper = shallow(
        <ClusterDetailsContainer
          clusterList={mockData.clusters}
          locale={locale}
          clusterDetailsContainerControl={
            mockData.clusterDetailsContainerControl
          }
        />
      ))
  );

  it("work as expected", () => {
    const instance = wrapper.instance();
    instance.handleSelection({}, "vbirsan1-remote");
    instance.handleSelection({}, undefined);
    instance.handleSelectionClear();
    instance.handleFirstClick();
    instance.handleLastClick();
    instance.handleNextClick({}, 1);
    instance.handlePreviousClick({}, 1);
    instance.handlePageInput({}, 1);
    instance.handleKeyPress(
      {
        action: "open_link",
        targetLink: "https://test"
      },
      {
        key: "Enter"
      }
    );
    instance.handleKeyPress(
      {
        action: "open_link",
        targetLink: "https://test"
      },
      {
        key: "Any"
      }
    );
    instance.handleSelectToggle();
    instance.handleExpandSectionToggle(0);
  });
});
