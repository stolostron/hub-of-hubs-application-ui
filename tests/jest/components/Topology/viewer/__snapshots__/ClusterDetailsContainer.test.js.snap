// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClusterDetailsContainer with no clusters renders as expected 1`] = `
<div
  className="clusterDetails"
>
  <div
    className="pf-c-select"
    data-ouia-component-id="OUIA-Generated-Select-typeahead-1"
    data-ouia-component-type="PF4/Select"
    data-ouia-safe={true}
  >
    <div
      className="pf-c-select__toggle pf-m-typeahead"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <div
        className="pf-c-select__toggle-wrapper"
      >
        <input
          aria-activedescendant={null}
          aria-label="Find cluster"
          autoComplete="off"
          className="pf-c-form-control pf-c-select__toggle-typeahead"
          disabled={false}
          id="pf-select-toggle-id-0-select-typeahead"
          onChange={[Function]}
          onClick={[Function]}
          placeholder="Find cluster"
          type="text"
          value=""
        />
      </div>
      <button
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-label="Options menu"
        aria-labelledby="cluster-select-id-1 pf-select-toggle-id-0"
        className="pf-c-button pf-c-select__toggle-button pf-m-plain"
        disabled={false}
        id="pf-select-toggle-id-0"
        onClick={[Function]}
        tabIndex={-1}
        type="button"
      >
        <svg
          aria-hidden={true}
          aria-labelledby={null}
          className="pf-c-select__toggle-arrow"
          fill="currentColor"
          height="1em"
          role="img"
          style={
            Object {
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 320 512"
          width="1em"
        >
          <path
            d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    className="spacer"
  />
  <div
    className="spacer"
  />
  <dl
    aria-label=""
    className="pf-c-accordion"
  />
</div>
`;

exports[`ClusterDetailsContainer with some clusters renders as expected 1`] = `
<div
  className="clusterDetails"
>
  <div
    className="pf-c-select"
    data-ouia-component-id="OUIA-Generated-Select-typeahead-2"
    data-ouia-component-type="PF4/Select"
    data-ouia-safe={true}
  >
    <div
      className="pf-c-select__toggle pf-m-typeahead"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <div
        className="pf-c-select__toggle-wrapper"
      >
        <input
          aria-activedescendant={null}
          aria-label="Find cluster"
          autoComplete="off"
          className="pf-c-form-control pf-c-select__toggle-typeahead"
          disabled={false}
          id="pf-select-toggle-id-1-select-typeahead"
          onChange={[Function]}
          onClick={[Function]}
          placeholder="Find cluster"
          type="text"
          value=""
        />
      </div>
      <button
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-label="Options menu"
        aria-labelledby="cluster-select-id-1 pf-select-toggle-id-1"
        className="pf-c-button pf-c-select__toggle-button pf-m-plain"
        disabled={false}
        id="pf-select-toggle-id-1"
        onClick={[Function]}
        tabIndex={-1}
        type="button"
      >
        <svg
          aria-hidden={true}
          aria-labelledby={null}
          className="pf-c-select__toggle-arrow"
          fill="currentColor"
          height="1em"
          role="img"
          style={
            Object {
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 320 512"
          width="1em"
        >
          <path
            d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    className="spacer"
  />
  <div
    className="spacer"
  />
  <dl
    aria-label=""
    className="pf-c-accordion"
  >
    <div
      className="clusterDetailItem"
      style={
        Object {
          "borderBottom": "1px solid #D2D2D2",
          "borderTop": "1px solid #D2D2D2",
        }
      }
    >
      <dt>
        <button
          aria-expanded={false}
          className="pf-c-accordion__toggle"
          id="argo-fxiang-eks"
          onClick={[Function]}
          type="button"
        >
          <span
            className="pf-c-accordion__toggle-text"
          >
            <svg
              fill="#3E8635"
              height="12px"
              width="12px"
            >
              <use
                className="label-icon"
                href="#diagramIcons_checkmark"
              />
            </svg>
            <span
              style={
                Object {
                  "paddingRight": "10px",
                }
              }
            />
            argo-fxiang-eks
          </span>
          <span
            className="pf-c-accordion__toggle-icon"
          >
            <svg
              aria-hidden={true}
              aria-labelledby={null}
              fill="currentColor"
              height="1em"
              role="img"
              style={
                Object {
                  "verticalAlign": "-0.125em",
                }
              }
              viewBox="0 0 256 512"
              width="1em"
            >
              <path
                d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
              />
            </svg>
          </span>
        </button>
      </dt>
      <dd
        aria-label=""
        className="pf-c-accordion__expanded-content"
        hidden={true}
        id=""
      >
        <div
          className="pf-c-accordion__expanded-content-body"
        >
          <span
            style={
              Object {
                "color": "#5A6872",
                "display": "block",
                "fontFamily": "RedHatText",
                "fontSize": "12px",
                "lineHeight": "21px",
                "textAlign": "left",
              }
            }
          >
            Namespace: undefined
          </span>
          <span
            className="label sectionLabel"
            style={
              Object {
                "fontSize": "1rem",
                "paddingLeft": "1rem",
              }
            }
          >
            Details
          </span>
          <div
            className="spacer"
          />
          <div
            className="sectionContent borderLeft"
          >
            <span
              className="label sectionLabel"
            >
              Name
              :
               
            </span>
            <span
              className="value"
            >
              argo-fxiang-eks
            </span>
          </div>
          <div
            className="sectionContent borderLeft"
          >
            <span
              className="label sectionLabel"
            >
              Namespace
              :
               
            </span>
            <span
              className="value"
            />
          </div>
          <div
            className="sectionContent borderLeft"
          >
            <span
              className="label sectionLabel"
            >
              Status
              :
               
            </span>
            <span
              className="value"
            >
              ok
            </span>
          </div>
          <div
            className="sectionContent borderLeft"
          >
            <span
              className="label sectionLabel"
            >
              Created
              :
               
            </span>
            <span
              className="value"
            >
              -
            </span>
          </div>
          <div
            className="spacer"
          />
        </div>
      </dd>
      <span
        style={
          Object {
            "color": "#5A6872",
            "display": "block",
            "fontFamily": "RedHatText",
            "fontSize": "12px",
            "lineHeight": "21px",
            "textAlign": "left",
          }
        }
      >
        Namespace: undefined
      </span>
    </div>
    <div
      className="clusterDetailItem"
      style={
        Object {
          "borderBottom": "1px solid #D2D2D2",
        }
      }
    >
      <dt>
        <button
          aria-expanded={false}
          className="pf-c-accordion__toggle"
          id="fxiang-eks"
          onClick={[Function]}
          type="button"
        >
          <span
            className="pf-c-accordion__toggle-text"
          >
            <svg
              fill="#3E8635"
              height="12px"
              width="12px"
            >
              <use
                className="label-icon"
                href="#diagramIcons_checkmark"
              />
            </svg>
            <span
              style={
                Object {
                  "paddingRight": "10px",
                }
              }
            />
            fxiang-eks
          </span>
          <span
            className="pf-c-accordion__toggle-icon"
          >
            <svg
              aria-hidden={true}
              aria-labelledby={null}
              fill="currentColor"
              height="1em"
              role="img"
              style={
                Object {
                  "verticalAlign": "-0.125em",
                }
              }
              viewBox="0 0 256 512"
              width="1em"
            >
              <path
                d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
              />
            </svg>
          </span>
        </button>
      </dt>
      <dd
        aria-label=""
        className="pf-c-accordion__expanded-content"
        hidden={true}
        id=""
      >
        <div
          className="pf-c-accordion__expanded-content-body"
        >
          <span
            style={
              Object {
                "color": "#5A6872",
                "display": "block",
                "fontFamily": "RedHatText",
                "fontSize": "12px",
                "lineHeight": "21px",
                "textAlign": "left",
              }
            }
          >
            Namespace: fxiang-eks
          </span>
          <span
            className="label sectionLabel"
            style={
              Object {
                "fontSize": "1rem",
                "paddingLeft": "1rem",
              }
            }
          >
            Details
          </span>
          <div
            className="spacer"
          />
          <div
            className="sectionContent borderLeft"
          >
            <span
              className="label sectionLabel"
            >
              Name
              :
               
            </span>
            <span
              className="value"
            >
              fxiang-eks
            </span>
          </div>
          <div
            className="sectionContent borderLeft"
          >
            <span
              className="label sectionLabel"
            >
              Namespace
              :
               
            </span>
            <span
              className="value"
            >
              fxiang-eks
            </span>
          </div>
          <div
            className="sectionContent borderLeft"
          >
            <span
              className="label sectionLabel"
            >
              Status
              :
               
            </span>
            <span
              className="value"
            >
              ok
            </span>
          </div>
          <div
            className="sectionContent borderLeft"
          >
            <span
              className="label sectionLabel"
            >
              CPU
              :
               
            </span>
            <span
              className="value"
            >
              3
              %
            </span>
          </div>
          <div
            className="sectionContent borderLeft"
          >
            <span
              className="label sectionLabel"
            >
              Memory
              :
               
            </span>
            <span
              className="value"
            >
              13
              %
            </span>
          </div>
          <div
            className="sectionContent borderLeft"
          >
            <span
              className="label sectionLabel"
            >
              Created
              :
               
            </span>
            <span
              className="value"
            >
              -
            </span>
          </div>
          <div
            className="spacer"
          />
        </div>
      </dd>
      <span
        style={
          Object {
            "color": "#5A6872",
            "display": "block",
            "fontFamily": "RedHatText",
            "fontSize": "12px",
            "lineHeight": "21px",
            "textAlign": "left",
          }
        }
      >
        Namespace: fxiang-eks
      </span>
    </div>
    <div
      className="clusterDetailItem"
      style={
        Object {
          "borderBottom": "1px solid #D2D2D2",
        }
      }
    >
      <dt>
        <button
          aria-expanded={false}
          className="pf-c-accordion__toggle"
          id="vbirsan1-remote"
          onClick={[Function]}
          type="button"
        >
          <span
            className="pf-c-accordion__toggle-text"
          >
            <svg
              fill="#3E8635"
              height="12px"
              width="12px"
            >
              <use
                className="label-icon"
                href="#diagramIcons_checkmark"
              />
            </svg>
            <span
              style={
                Object {
                  "paddingRight": "10px",
                }
              }
            />
            vbirsan1-remote
          </span>
          <span
            className="pf-c-accordion__toggle-icon"
          >
            <svg
              aria-hidden={true}
              aria-labelledby={null}
              fill="currentColor"
              height="1em"
              role="img"
              style={
                Object {
                  "verticalAlign": "-0.125em",
                }
              }
              viewBox="0 0 256 512"
              width="1em"
            >
              <path
                d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
              />
            </svg>
          </span>
        </button>
      </dt>
      <dd
        aria-label=""
        className="pf-c-accordion__expanded-content"
        hidden={true}
        id=""
      >
        <div
          className="pf-c-accordion__expanded-content-body"
        >
          <span
            style={
              Object {
                "color": "#5A6872",
                "display": "block",
                "fontFamily": "RedHatText",
                "fontSize": "12px",
                "lineHeight": "21px",
                "textAlign": "left",
              }
            }
          >
            Namespace: vbirsan1-remote
          </span>
          <span
            className="label sectionLabel"
            style={
              Object {
                "fontSize": "1rem",
                "paddingLeft": "1rem",
              }
            }
          >
            Details
          </span>
          <div
            className="spacer"
          />
          <div
            className="sectionContent borderLeft"
          >
            <span
              className="label sectionLabel"
            >
              Name
              :
               
            </span>
            <span
              className="value"
            >
              vbirsan1-remote
            </span>
          </div>
          <div
            className="sectionContent borderLeft"
          >
            <span
              className="label sectionLabel"
            >
              Namespace
              :
               
            </span>
            <span
              className="value"
            >
              vbirsan1-remote
            </span>
          </div>
          <div
            className="sectionContent borderLeft"
          >
            <span
              className="link sectionLabel"
              id="linkForNodeAction"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex="0"
            >
              Open cluster console
              <svg
                height="12px"
                style={
                  Object {
                    "marginLeft": "8px",
                    "stroke": "#0066CC",
                  }
                }
                width="12px"
              >
                <use
                  className="label-icon"
                  href="#diagramIcons_carbonLaunch"
                />
              </svg>
            </span>
          </div>
          <div
            className="sectionContent borderLeft"
          >
            <span
              className="label sectionLabel"
            >
              Status
              :
               
            </span>
            <span
              className="value"
            >
              ok
            </span>
          </div>
          <div
            className="sectionContent borderLeft"
          >
            <span
              className="label sectionLabel"
            >
              CPU
              :
               
            </span>
            <span
              className="value"
            >
              8
              %
            </span>
          </div>
          <div
            className="sectionContent borderLeft"
          >
            <span
              className="label sectionLabel"
            >
              Memory
              :
               
            </span>
            <span
              className="value"
            >
              4
              %
            </span>
          </div>
          <div
            className="sectionContent borderLeft"
          >
            <span
              className="label sectionLabel"
            >
              Created
              :
               
            </span>
            <span
              className="value"
            >
              -
            </span>
          </div>
          <div
            className="spacer"
          />
        </div>
      </dd>
      <span
        style={
          Object {
            "color": "#5A6872",
            "display": "block",
            "fontFamily": "RedHatText",
            "fontSize": "12px",
            "lineHeight": "21px",
            "textAlign": "left",
          }
        }
      >
        Namespace: vbirsan1-remote
      </span>
    </div>
  </dl>
</div>
`;
